Flaky test on mindeps: `test_concat_categorical`

This test frequently fails on the `mindeps-optional` environment only:
https://github.com/dask/dask/actions/runs/21628536686/job/62334916047?pr=12266

```
FAILED dask/dataframe/tests/test_multi.py::test_concat_categorical[True-False-True] - AssertionError: Series are different

Series values are different (26.66667 %)
[index]: [0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4]
[left]:  ['a', 'a', 'b', 'b', 'b', ..., 'b', 'b', 'a', 'c', 'c']
Length: 15
Categories (3, object): ['a', 'b', 'c']
[right]: ['a', 'a', 'b', 'b', 'c', ..., 'b', 'b', 'a', 'c', 'c']
Length: 15
Categories (3, object): ['a', 'b', 'c']
At positional index 4, first diff: b != c
FAILED dask/dataframe/tests/test_multi.py::test_concat_categorical[False-False-True] - AssertionError: Series are different

Series values are different (26.66667 %)
[index]: [0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4]
[left]:  ['a', 'a', 'b', 'b', 'b', ..., 'b', 'b', 'a', 'c', 'c']
Length: 15
Categories (3, object): ['a', 'b', 'c']
[right]: ['a', 'a', 'b', 'b', 'c', ..., 'b', 'b', 'a', 'c', 'c']
Length: 15
Categories (3, object): ['a', 'b', 'c']
At positional index 4, first diff: b != c
```

--- a/dask/dataframe/tests/test_multi.py
+++ b/dask/dataframe/tests/test_multi.py
@@ -1907,7 +1907,7 @@ def test_concat5():
                 PANDAS_GE_220,
                 reason="https://github.com/dask/dask/issues/10558",
                 raises=AssertionError,
-                strict=True,
+                strict=False,
             ),
         ),
         (True, False, False),
@@ -1920,14 +1920,13 @@ def test_concat5():
                 PANDAS_GE_220,
                 reason="https://github.com/dask/dask/issues/10558",
                 raises=AssertionError,
-                strict=True,
+                strict=False,
             ),
         ),
         (False, False, False),
     ],
 )
-@pytest.mark.parametrize("repeat", range(100))
-def test_concat_categorical(known, cat_index, divisions, repeat):
+def test_concat_categorical(known, cat_index, divisions):
     frames = [
         pd.DataFrame(
             {

